Loop/Views/SuspendThresholdPicker.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  SuspendThresholdPicker.swift
+//  Loop
+//
+//  Created by Michael Pangburn on 4/17/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+import HealthKit
+import LoopKit
+import LoopKitUI
+
+
+struct SuspendThresholdPicker: View {
+    @Binding var value: HKQuantity
+    var unit: HKUnit
+    @Binding var isEditing: Bool
+
+    private var formatter: NumberFormatter
+
+    init(value: Binding<HKQuantity>, unit: HKUnit, isEditing: Binding<Bool>) {
+        self._value = value
+        self.unit = unit
+        self._isEditing = isEditing
+        self.formatter = {
+            let quantityFormatter = QuantityFormatter()
+            quantityFormatter.setPreferredNumberFormatter(for: unit)
+            return quantityFormatter.numberFormatter
+        }()
+    }
+
+    var body: some View {
+        VStack {
+            HStack {
+                Spacer()
+
+                if Guardrail.suspendThreshold.classification(for: value) != .withinRecommendedRange {
+                    Image(systemName: "exclamationmark.triangle.fill")
+                        .foregroundColor(accentColor)
+                        .transition(.springInScaleOut)
+                }
+
+                Text(formatter.string(from: value.doubleValue(for: unit)) ?? "\(value.doubleValue(for: unit))")
+                    .foregroundColor(accentColor)
+                    .animation(nil)
+
+                Text(unit.shortLocalizedUnitString())
+                    .foregroundColor(.gray)
+            }
+            .onTapGesture {
+                withAnimation {
+                    self.isEditing.toggle()
+                }
+            }
+
+            if isEditing {
+                GlucoseValuePicker(value: $value.animation(), unit: unit, guardrail: .suspendThreshold)
+                    .padding(.horizontal, -8)
+                    .transition(.expandFromTop)
+            }
```

Comment:
No issues, but do you think one might want a more generic picker that includes some view builder for display of selected value while providing all this animation, gesture recognition, unit placement, etc? I mean, this pairing between the picker and display of selected value is probably something that will be encountered again and again. Definitely a selfish request.

Reply:
Yeah, I've found a similar need in continuing to build out other guardrails. I'm also going to generalize `GlucoseValuePicker` to `QuantityValuePicker`—will do these in a follow-up LoopKit PR.